src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -212,7 +213,7 @@ public void apply(final AssemblyRegion region, final ReferenceContext referenceC
         hcEngine.callRegion(region, featureContext).stream()
                 // Only include calls that start within the current read shard (as opposed to the padded regions around it).
                 // This is critical to avoid duplicating events that span shard boundaries!
-                .filter(call -> getCurrentReadShardBounds().contains(call))
+                .filter(call -> getCurrentReadShardBounds().contains(new SimpleInterval(call.getContig(), call.getStart(), call.getStart())))
```

Comment:
If I understand the change correctly, this fixes previous behavior where large insertions starting within one read shard and ending in another would be filtered, and replaces it with what the comment says to do.  Seems right to me.